# 3.1 Data Collection

### Problem Statement

Pull contents of subreddits- spacex and blueorigin using Reddit API and then use NLP to train a classifier on which subreddit a given post came from. 

### Data Collection

In [1]:
import requests
import time

In [2]:
sub_reddit_1 = 'spacex'
sub_reddit_2 = 'blueorigin'
url_temp = 'https://www.reddit.com/r/{}.json'
headers = {'User-agent': 'Bleep blorp bot 0.1'}

In [3]:
# create a function to pull all titles of the post
# since most reddit posts only have titles
def get_all_posts(sub_reddit):
    after = ''
    all_posts = []
    while len(all_posts) < 1200:
        if after:
#             print('after is {}'.format(after))
            params = {'after': after}
        else:
            params = {}
        response = requests.get(url_temp.format(sub_reddit), headers=headers, params=params)
        after = response.json()['data']['after']
        posts = response.json()['data']['children']
        for post in posts:
            if 'data' in post:
                all_posts.append(post['data']['title'])
#         print("{} posts saved!".format(len(all_posts)))
        time.sleep(5)
    return all_posts

In [4]:
# pull posts for spacex
all_posts_1 = get_all_posts(sub_reddit_1)

In [5]:
len(all_posts_1)

1216

In [12]:
# remove duplicated posts
unique_posts_1 = set(all_posts_1)

In [13]:
len(unique_posts_1)

989

In [6]:
# pull posts for blueorigin
all_posts_2 = get_all_posts(sub_reddit_2)

In [9]:
len(all_posts_2)

1210

In [14]:
# remove duplicated posts
unique_posts_2 = set(all_posts_2)

In [15]:
len(unique_posts_2)

977

Pulled 989 posts for spacex and 977 posts for blueorigin.

### Save Data as csv

In [19]:
import pandas as pd

In [23]:
# convert data to dataframe and save as csv
subreddit_1 = pd.DataFrame(unique_posts_1)
subreddit_1.to_csv('./datasets/subreddit_1.csv', index=False)

In [24]:
# convert data to dataframe and save as csv
subreddit_2 = pd.DataFrame(unique_posts_2)
subreddit_2.to_csv('./datasets/subreddit_2.csv', index=False)